tests/func/remote/test_gdrive.py

```diff
@@ -0,0 +1,56 @@
+import os
+import posixpath
+
+import configobj
+
+from dvc.compat import fspath
+from dvc.main import main
+from dvc.remote import GDriveRemote
+from dvc.repo import Repo
+
+
+def test_relative_user_credentials_file_config_setting(tmp_dir, dvc):
+    # CI sets it to test GDrive, here we want to test the work with file system
+    # based, regular credentials
+    if os.getenv(GDriveRemote.GDRIVE_CREDENTIALS_DATA):
+        del os.environ[GDriveRemote.GDRIVE_CREDENTIALS_DATA]
+
+    credentials = os.path.join("secrets", "credentials.json")
+
+    # GDriveRemote.credentials_location helper checks for file existence,
+    # create the file
+    tmp_dir.gen(credentials, "{'token': 'test'}")
+
+    remote_name = "gdrive"
+    assert (
+        main(["remote", "add", "-d", remote_name, "gdrive://root/test"]) == 0
+    )
+    assert (
+        main(
+            [
+                "remote",
+                "modify",
+                remote_name,
+                "gdrive_user_credentials_file",
+                credentials,
+            ]
+        )
+        == 0
+    )
```

Comment:
Let's use api instead of CLI. Just makes it so much cleaner. See `dvc/command/remote.py`

Reply:
Was thinking since it's functional it's better to test "real" scenario ... API is _almost_ the same, but not exactly.

Reply:
functional doesn't mean CLI. We try to use API in functional tests and just simply unit-test CLI, because CLI doesn't have much logic in it. In this PR you don't really change `dvc remote` CLI, but rather config API, so there isn't much sense in using CLI here.

Reply:
do you mean something like:

```py
with self.config.edit() as conf:
     conf["remote"][name] = {url ... gdrive_...}
```

I had this initially :)

really, really don't like this idea. It is not functional to me. I don't see an actual Repo API to do these operations.

Reply:
@shcheklein Then your test is out-of-place here. And maybe it is actually. You intent on testing that config properly resolves your paths, but in the test you touch a whole lot of things. Consider removing this test and writing a unit test for config object to make sure that it properly resolves `gdrive_user_credentials_file`. That would be much nicer.

Reply:
I'm testing this scenario:

```
dvc add remote gdrive
dvc add remote modify gdrive_user_credentials_file relative_path
# path is saved properly
dcv <any command that uses gdrive_user_credentials_file> <--- make sure that it reads it properly
```

this is exactly _functional_ flow behavior from the end-user perspective. That's how `gdrive_user_credentials_file` is documented more or less.

Reply:
Yeah, but the changes you've introduced only seem to affect config. There is no real need to run it through the whole CLI stack. But could keep it as is, better than no test. Your call.

Reply:
👍